app/assets/javascripts/code/services/_mobile_sessions.js

```diff
@@ -81,26 +83,34 @@ export const mobileSessions = (
     },
 
     selectSession: function(id) {
+      drawSession.clear(this.sessions);
       const callback = (session, allSelected) => (data) => {
         prevMapPosition = {
           bounds: map.getBounds(),
           zoom: map.getZoom()
         };
-        const draw = () => drawSession.drawMobileSession(session, boundsCalculator(allSelected));
+        const draw = () => drawSession.drawMobileSession(session);
         map.fitBounds(boundsCalculator(allSelected));
         sessionsUtils.onSingleSessionFetch(session, data, draw);
       }
       this._selectSession(id, callback);
     },
 
     reSelectSession: function(id) {
+      // I think this is never used since we blocked the button to select all sessions
+      drawSession.clear(this.sessions);
       const callback = (session, allSelected) => (data) => {
-        const draw = () => drawSession.drawMobileSession(session, boundsCalculator(allSelected));
+        const draw = () => drawSession.drawMobileSession(session);
         sessionsUtils.onSingleSessionFetch(session, data, draw);
       }
       this._selectSession(id, callback);
     },
 
+    drawSessionsInLocation: function() {
```

Comment:
This is really similar to the fixed one.

Also, keep in mind the `sessionUtils` refactor was mine and I hate myself for doing it. I should have eliminated the `this` instead of passing it to `sessionsUtils`. The idea was to keep there the common code.

Reply:
I changed and it's different from fixed now. But we should probably visit and refactor fixed someday as well, then we can also extract similar parts to utils.